src/EditorFeatures/Core.Wpf/InlineErrors/InlineErrorTaggerProvider.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Implementation.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Implementation.EditAndContinue;
+using Microsoft.CodeAnalysis.Editor.Shared.Options;
+using Microsoft.CodeAnalysis.Editor.Shared.Tagging;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Editor.Tagging;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Adornments;
+using Microsoft.VisualStudio.Text.Classification;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+using Microsoft.VisualStudio.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.InlineErrors
+{
+    [Export(typeof(ITaggerProvider))]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    [ContentType(ContentTypeNames.XamlContentType)]
```

Comment:
interesting.  does xaml want this?

Reply:
Not sure. What does it do?

Reply:
I'd say probably not to start with. We can enable for XAML later if we'd like :)

Reply:
In general, our customers would like the experience in XAML to be closer to C#, so from that perspective I would be inclined to keep it for XAML.

But, it seems that the option to turn this on/off would be under C# so that presents a problem for other languages. I'm also concerned about the experience when the lines are longer. Some more screenshots with long lines and the different options would be great.

I wonder if these should behave more like code lens and "make space" above for example. Also, what happens if you have multiple errors in one line?

Reply:
I am fully on board with enabling for XAML, I just think there is non-zero effort to do that and this PR should not be burdened with that. We want it to be a good experience, and this PR focuses on the C#/VB side of things. Once we nail that I think we can expand to XAML and see what that takes.